pkg/agent/route/route_linux.go

```diff
@@ -1246,29 +1234,41 @@ func (c *Client) AddClusterIPRoute(svcIP net.IP) error {
 	// Collect stale routes.
 	var staleRoutes []*netlink.Route
 	if curClusterIPCIDR != nil {
-		// If current destination CIDR is not nil, the route with current destination CIDR should be uninstalled.
+		// If the current ClusterIP CIDR exists, the route with the current ClusterIP CIDR should be uninstalled since
+		// a new route with a newly calculated ClusterIP CIDR has been installed.
 		route.Dst = curClusterIPCIDR
 		staleRoutes = []*netlink.Route{route}
 	} else {
-		// If current destination CIDR is nil, which means that Antrea Agent has just started, then all existing routes
-		// whose destination CIDR contains the first ClusterIP should be uninstalled, except the newly installed route.
-		// Note that, there may be multiple stale routes prior to this commit. When upgrading, all stale routes will be
-		// collected. After this commit, there will be only one stale route after Antrea Agent started.
 		routes, err := c.listIPRoutesOnGW()
 		if err != nil {
 			return fmt.Errorf("error listing ip routes: %w", err)
 		}
+		// If current ClusterIP CIDR doesn't exist, which means that Antrea Agent has just started, and the route for
+		// ClusterIP generated in the last round should be deleted.
 		for i := 0; i < len(routes); i++ {
-			if routes[i].Gw.Equal(gw) && !routes[i].Dst.IP.Equal(svcIP) && routes[i].Dst.Contains(svcIP) {
+			// Skip the route whose destination CIDR only has one address.
```

Comment:
Why skip the routing entry for a single IP address?

Reply:
The routing entry for a single IP address (mask is 32 or 128) could be for LoadBalancer or the first ClusterIP, and such routing entry should be reserved.

Reply:
The code comment is unclear and puzzling.
And I feel the above explanation still doesn't make sense. It will never delete stale LoadBalancer IPs, and it doesn't make sense to handle first ClusterIP in this way.

Reply:
How about
```
// Skip the route whose destination CIDR has only one address since it could be the route for the first
// ClusterIP newly installed. Note that, this could also match a route for a LoadBalancer ingress IP, but
// it doesn't affect skipping the route for the first ClusterIP.
```